pybamm/expression_tree/averages.py

```diff
@@ -0,0 +1,366 @@
+        super().__init__(child, "size-average", integration_variable)
+        self.f_a_dist = f_a_dist
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return size_average(child, f_a_dist=self.f_a_dist)
+
+
+def x_average(symbol):
+    """
+    convenience function for creating an average in the x-direction.
+
+    Parameters
+    ----------
+    symbol : :class:`pybamm.Symbol`
+        The function to be averaged
+
+    Returns
+    -------
+    :class:`Symbol`
+        the new averaged symbol
+    """
+    # Can't take average if the symbol evaluates on edges
+    if symbol.evaluates_on_edges("primary"):
+        raise ValueError("Can't take the x-average of a symbol that evaluates on edges")
+    # If symbol doesn't have an electrode domain, its x-averaged value is itself
+    if not any(
+        any(
+            dom in ["negative electrode", "separator", "positive electrode"]
+            for dom in domain
+        )
+        for domain in symbol.domains.values()
+    ):
+        new_symbol = symbol.new_copy()
+        new_symbol.parent = None
+        return new_symbol
+    # If symbol is a broadcast, reduce by one dimension
+    if isinstance(
+        symbol,
+        (pybamm.PrimaryBroadcast, pybamm.SecondaryBroadcast, pybamm.FullBroadcast),
+    ):
+        if all(
+            dom in ["negative electrode", "separator", "positive electrode"]
+            for dom in symbol.broadcast_domain
+        ):
+            return symbol.reduce_one_dimension()
+        elif isinstance(symbol, pybamm.PrimaryBroadcast):
+            return pybamm.PrimaryBroadcast(
+                x_average(symbol.orphans[0]), symbol.broadcast_domain
+            )
+        elif isinstance(symbol, pybamm.FullBroadcast) and all(
+            dom in ["negative electrode", "separator", "positive electrode"]
+            for dom in symbol.secondary_domain
+        ):
+            aux = {}
+            if "tertiary" in symbol.auxiliary_domains:
+                aux["secondary"] = symbol.auxiliary_domains["tertiary"]
+            return pybamm.FullBroadcast(symbol.orphans[0], symbol.broadcast_domain, aux)
+        elif (
+            isinstance(symbol, pybamm.FullBroadcast)
+            and "tertiary" in symbol.auxiliary_domains
+            and all(
+                dom in ["negative electrode", "separator", "positive electrode"]
+                for dom in symbol.tertiary_domain
+            )
+        ):
+            aux = {"secondary": symbol.auxiliary_domains["secondary"]}
+            if "quaternary" in symbol.auxiliary_domains:
+                aux["tertiary"] = symbol.auxiliary_domains["quaternary"]
+            return pybamm.FullBroadcast(symbol.orphans[0], symbol.broadcast_domain, aux)
+        else:  # pragma: no cover
+            # It should be impossible to get here
+            raise NotImplementedError
+    # If symbol is a concatenation of Broadcasts, its average value is its child
```

Comment:
Probably a very dumb question but I don't get why that is the case should you take the average of each child and then average them together? (maybe it is what you are doing already, but didn't understand it).

Reply:
That is what I'm doing, I'll change the comment